libs/components/auto-scale/auto-scale-img.directive.ts

```diff
@@ -0,0 +1,24 @@
+import { ChangeDetectorRef, Directive, ElementRef, HostListener, Self } from '@angular/core';
+
+@Directive({
+  selector: '[gewdAutoScaleImg]'
+})
+export class AutoScaleImgDirective {
+
+  @HostListener('load')
+  public onLoad() {
+    this._cd.markForCheck();
```

Comment:
I was wondering why the markForCheck works here - but probably because the parent component is not `OnPush` ?

Reply:
So in this scenario it would work without the change detection. I just noticed, that the CD would do anything, because it would only work tree downwards and I want the `AutoScaleComponent` to change. Thats why I changed it and added a method to update the component from the outside aswell so if you encounter any update problems in the future, you could just trigger CD manually from the outside. Check out my newly pushed updates.